_posts/2020-10-09-torch-optim/torch_network_with_optim.Rmd

```diff
@@ -0,0 +1,269 @@
+---
+title: "Optimizers in torch"
+description: > 
+ Today, we wrap up our mini-series on torch basics, adding to our toolset two abstractions we hadn't made use of before: loss functions and optimizers.
+
+author:
+  - name: Sigrid Keydana
+    affiliation: RStudio
+    affiliation_url: https://www.rstudio.com/
+slug: keydanatorchnetworkfromscratch
+date: 10-01-2020
+categories:
+  - Torch
+  - R
+bibliography: bibliography.bib
+output:
+  distill::distill_article:
+    self_contained: false
+    toc: true
+preview: images/preview.jpg
+---
+
+```{r setup, include=FALSE}
+knitr::opts_chunk$set(echo = TRUE, eval = FALSE)
+```
+
+```{r}
+
+```
+
+This is the fourth and last installment in a series introducing `torch` basics. Initially, we [focused on *tensors*](https://blogs.rstudio.com/ai/posts/2020-10-01-torch-network-from-scratch/). To illustrate their power, we coded a complete (if toy-size) neural network from scratch. We didn't make use of any of `torch`'s higher-level capabilities -- not even *autograd*, its automatic-differentiation feature.
```

Comment:
I would rework this to put the autograd description with the second post (e.g., In the follow-up post, we made use of autograd, torch's automatic differentiation feature…).